PYME/Acquire/Hardware/Piezos/piezo_c867.py

```diff
@@ -310,16 +334,18 @@ def close(self):
             #time.sleep(.01)
             #self.ser_port.close()            
                 
-        
-    def SetServo(self, state=1):
-        self.lock.acquire()
-        try:
-            self.ser_port.write(b'SVO 1 %d\n' % state)
-            self.ser_port.write(b'SVO 2 %d\n' % state)
-            self.servo = state == 1
-        finally:
-            self.lock.release()
-            
+    # directly enabling/disabling the servo outside the thread loop
+    #      seems to hang PYME
+    # the implementation below instead 'messages' the thread loop by setting flags
+    #      that get picked up in thread loop processing
+    # I dislike state being an int and rather have opted treating as bool which reflects the on/off state better IMHO
+    def SetServo(self, state=True):
+        if self.servo and not state: # need to toggle servo off
```

Comment:
I don't like setting two separate flags (which then get modified in the polling thread) for turning the servo on or off. This is neither thread safe, nor particularly clear.

a better approach would be to adopt the approach used for the positions, and just set a `._target_servo_state` variable when called outside the communications thread. Inside the thread, we could check if the current servo state (or a cached local variable recording the last commanded state) is the same as `_target_servo_state` and change the servo state if it is not. This is thread-safe, as the value only ever gets written from outside the communications thread.